lib/stylelint/plugin.rb

```diff
@@ -1,34 +1,124 @@
-  # the public interface documented. Danger uses [YARD](http://yardoc.org/)
-  # for generating documentation from your plugin source, and you can verify
-  # by running `danger plugins lint` or `bundle exec rake spec`.
-  #
-  # You should replace these comments with a public description of your library.
+  # Lint stylesheets using [stylelint](http://stylelint.io/).
+  # Results are send as inline comment.
   #
-  # @example Ensure people are well warned about merging on Mondays
+  # @example Run stylelint with changed files only
   #
-  #          my_plugin.warn_on_mondays
+  #          stylelint.changes_only = true
+  #          stylelint.lint
   #
-  # @see  Rossukhon Leagmongkol/danger-stylelint
-  # @tags monday, weekends, time, rattata
+  # @see  nimblehq/danger-stylelint
+  # @tags lint, stylelint, css, scss, sass
   #
   class DangerStylelint < Plugin
-    # An attribute that you can read/write from your Dangerfile
+    DEFAULT_BIN_PATH = "./node_modules/.bin/stylelint"
+
+    # An path to stylelint's config file
+    # @return [String]
+    attr_accessor :config_file
+
+    # Enable changes_only
+    # Only show messages within changed files.
+    # @return [Boolean]
+    attr_accessor :changes_only
+
+    # A path of stylelint's bin
+    # @return [String]
+    attr_writer :bin_path
+
+    def bin_path
+      @bin_path ||= DEFAULT_BIN_PATH
+    end
+
+    # Specify extensions of target file
+    # Default is [".css, .scss, .sass"]
+    # @return [Array]
+    attr_writer :target_extensions
+
+    def target_extensions
+      @target_extensions ||= %w(.css .scss .sass)
+    end
+
+    # Lints css files.
+    # Generates `errors` and `warnings` according to stylelint's config.
     #
-    # @return   [Array<String>]
-    attr_accessor :my_attribute
+    # @return  [void]
+    #
+    def lint
+      lint_results
+        .reject { |result| result.nil? || result["warnings"].length.zero? }
+        .map { |result| send_comment result }
+    end
 
-    # A method that you can call from your Dangerfile
-    # @return   [Array<String>]
+    private
+
+    # Get stylelint bin path
     #
-    def warn_on_mondays
-      warn "Trying to merge code on a Monday" if Date.today.wday == 1
+    # return [String]
+    def stylelint_path
+      raise "stylelint is not installed" unless File.exist?(bin_path)
+
+      bin_path
+    end
+
+    # Get all lintable files
+    #
+    # return [String]
+    def all_lintable_files
+      "**/*{#{target_extensions.join(',')}}"
+    end
+
+    # Get lint result with respect to the changes_only option
+    #
+    # return [Hash]
+    def lint_results
+      bin = stylelint_path
+      return run_lint(bin, all_lintable_files) unless changes_only
+
+      changed_files.select { |f| target_extensions.include?(File.extname(f)) }
+        .map { |f| f.gsub("#{Dir.pwd}/", "") }
+        .map { |f| run_lint(bin, f).first }
+    end
+
+    # Get all the changed files
+    #
+    # return [Array]
+    def changed_files
```

Comment:
Should we place this next to the `all_lintable_files`? it could be `all_changed_files` to be consistent.

Reply:
sure, done in de123cae1830f1defd5c96af06cb480faa4d1aec!